### PR TITLE
[ENH] added coverage parameter to all metrics of pred_interval type

### LINKED ISSUE #7276: [ENH] add coverage setting for EmpiricalCoverage metric

### PR DESCRIPTION
**Pull request by talat-khattatov**
Closes #7276

### CODE REVIEW

**Review by fkiraly — CHANGES_REQUESTED** [review]
Nice, thanks for the quick fix!

We should also add a test that this is correctly used.

I think the best way is to expand upon `test_evaluate_probabilistic` in `forecasting.model_evaluation.tests`, and check the `coverage` in the returned data, if `return_data=True`. Or do you have an idea how to test that the argument works?

**Comment by talat-khattatov** [comment]
tests are added

File: sktime/forecasting/model_evaluation/tests/test_evaluate.py [hunk 1]
```diff
@@ -358,6 +365,38 @@ def test_evaluate_probabilistic(n_columns, scoring):
...
+)
+@pytest.mark.parametrize("n_columns", [1, 2])
+@pytest.mark.parametrize("scoring", INTERVAL_METRICS_WITH_PARAMS)
+def test_evaluate_probabilistic_with_params(n_columns, scoring):
+    """Check that evaluate works with interval, quantile, and distribution forecasts."""
+    y = _make_series(n_columns=n_columns)
+
+    forecaster = NaiveForecaster()
+    cv = SlidingWindowSplitter()
+    try:
```
Comment (fkiraly) [inline sktime/forecasting/model_evaluation/tests/test_evaluate.py:380]: A very general recommendation: try/except should always be avoided in a case where one could to if/else instead, with a known condition.

I suspect in this case, we do know whne the `NotImplementedError` occurs? Could oyu kindly explain? Can we change this to if/else?

**Review by fkiraly — CHANGES_REQUESTED** [review]
Nice! Should be good to go - almost. I added a remark above about the try/except, could we resolve that?

File: sktime/forecasting/model_evaluation/tests/test_evaluate.py (hunk 1 above)
Comment (talat-khattatov) [reply sktime/forecasting/model_evaluation/tests/test_evaluate.py:380]: Thanks. It is indeed redundant. I copy pasted the skeleton of the test from the test above **test_evaluate_probabilistic** . I will remove it. Does not affect the test results.

**Review by fkiraly — APPROVED** [review]

**Review by fkiraly — APPROVED** [review]
Great! Also thanks for adding more estimators to the tests.
I have also updated the docstrings to make usage clearer.

Btw, there are two further things you could help with if you want:

* the same location exists in `skpro` - this is on track to be refactored to a common location, the metrics would eventually move there. It would be great if you could update it there to, the recipe is copy-paste. Later, we will refactor that there is only one location.
* The tests for the proba metrics do not follow the common pattern of the `TestAll` classes - this should also change in the mid term. This is a bit more involved but would be a great piece of work. Issue is here: https://github.com/sktime/sktime/issues/4307

**Pull request merged**